### PR TITLE
Strict Null Checks - Fix `SanitizedIntegrationConfig.id` & PixieBrix integration handling

### PR DESCRIPTION
**Pull request by BLoe**
## What does this PR do?

- Unblocks some strict-null-checks work
- Fixes pixiebrix integration configs so they don't have and undefined id value
- Refactors some of the integration config handling to untangle app user behavior checking ("no config selected") from the background config locator

## Remaining work

- [x] Make sure pixiebrix integrations are working properly in the page editor etc.

## Checklist

- [x] Add jest or playwright tests and/or storybook stories
- [x] Designate a primary reviewer - @grahamlangford

### CODE REVIEW

**Comment by BLoe** [comment]
I'm not totally sure why the end to end tests are failing, I'll get them running locally and figure out if there is a real problem or not. 🤷‍♂️

File: src/contrib/google/sheets/core/useSpreadsheetId.ts [hunk 1]
```diff
@@ -79,15 +80,23 @@ async function findSpreadsheetIdFromFieldValue(
...
     );
   }
 
   const { integrationId, configId } = sheetIdIntegrationDependency;
+
+  if (!configId) {
+    throw new BusinessError(
+      "No configuration selected for the Google Sheets integration",
+    );
+  }
```
Comment (grahamlangford) [inline src/contrib/google/sheets/core/useSpreadsheetId.ts:94]: We have an `assertNotNullish` utility, though it throws a `TypeError`

File: src/integrations/integrationTypes.ts [hunk 2]
```diff
@@ -186,8 +186,7 @@ export interface SanitizedIntegrationConfig {
   /**
    * UUID of the integration configuration.
    */
-  // ID is only undefined for the PixieBrix integration
-  id?: UUID;
+  id: UUID;
```
Comment (grahamlangford) [inline src/integrations/integrationTypes.ts:189]: 💜

File: src/integrations/locator.ts [hunk 3]
```diff
@@ -268,28 +256,22 @@ class LazyLocatorFactory {
...
 
-    if (!this.initialized) {
-      await this.refresh();
-    }
-
     if (serviceId === PIXIEBRIX_INTEGRATION_ID) {
-      // HACK: for now use the separate storage for the extension key
-      return pixiebrixConfigurationFactory();
+      // Since 1.8.13 the locator should not be used to instantiate the pixiebrix integration config
+      throw new Error(
```
Comment (grahamlangford) [inline src/integrations/locator.ts:268]: Should this be a `BusinessError`?

**Review by grahamlangford — APPROVED** [review]
Left some minor comments. But overall looks good

**Pull request merged**